src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -113,15 +117,26 @@ public ValueTask<FlushResult> WriteStreamSuffixAsync()
         {
             lock (_contextLock)
             {
-                if (_suffixSent || !_autoChunk)
+                if (_suffixSent)
+                {
+                    // If WriteStreamSuffixAsync has already beend called, no-op and return the previously returned ValueTask.
```

Comment:
nit: return _writeStreamSuffxiValueTask here to avoid setting _suffixSent to true every time

Reply:
It shouldn't be a problem to set it _suffixSent true every time. I can rename it though, since Content-Length responses don't actually have a suffix.

Reply:
Isn't it needless though to set it to true every time? Would the compiler optimize it out?

Also empty if blocks are ehhhhh.

Reply:
I had the early return first, but it looked worse in my opinion.

Reply:
I changed it for you though.